src/os/bluestore/BlueStore.h

```diff
@@ -2728,4 +2752,149 @@ static inline void intrusive_ptr_release(BlueStore::OpSequencer *o) {
   o->put();
 }
 
+class BlueStoreRepairer
+{
+public:
+  // to simplify future potential migration to mempools
+  using fsck_interval = interval_set<uint64_t>;
+
+  // Structure to track cid/oid -> pextent mapping.
+  // Similar to Bloom filter positive and false-positive matches are 
+  // possible only.
+  // Differs from the BF as matches are verified on CID or OID only.
+  // Allows to filter out 'uninteresting' pextents prior to match verification.
```

Comment:
Some more detail on how this data structure is layed out would be helpful.  I'm still reading the code but having trouble understanding why the bmaps are sharded out by cid, and why.

Reply:
OK, will do. Actually that's simply a way to answer if specific CID might be using specific space range. Store space is split into areas according to calculated granularity and for each area corresponding (sub-)bitmap (with bit per shard, 256 bits total for now) determines whether CIDs (belonging to a shard) reference the area. CID to shard mapping is just a hash(CID). 
The same standalone procedure for OIDs.